### ISSUE TITLE
0011450:
caldav sync with thunderbird stopped working

### ISSUE THREAD

**Issue by Gloirin** [tine20/Tine-2.0-Open-Source-Groupware-and-CRM#5665]
**Reported by weberho on 20 Nov 2015 10:14**

**Version:** Elena (2015.07.6)

With the latest update caldav sync with Thunderbird broke. IMO the issue might be related to the changes in #11428. The problem seem to be in the fourth request thunderbird sends.

REQUEST: PROPFIND /calendars/2/12/ HTTP/1.1&quot; 207

&lt;?xml version=&quot;1.0&quot; encoding=&quot;utf-8&quot;?&gt;
&lt;d:multistatus xmlns:d=&quot;DAV:&quot; xmlns:s=&quot;http://sabredav.org/ns&quot; xmlns:card=&quot;urn:ietf:params:xml:ns:carddav&quot; xmlns:cal=&quot;urn:ietf:params:xml:ns:caldav&quot; xmlns:cs=&quot;http://calendarserver.org/ns/&quot; xmlns:owncloud=&quot;http://owncloud.org/ns&quot;&gt;&lt;d:response&gt;&lt;d:href&gt;/calendars/2/12/&lt;/d:href&gt;&lt;d:propstat&gt;&lt;d:prop&gt;&lt;d:current-user-principal&gt;&lt;d:href&gt;/principals/users/2/&lt;/d:href&gt;&lt;/d:current-user-principal&gt;&lt;d:owner&gt;&lt;d:href&gt;/principals/users/2/&lt;/d:href&gt;&lt;/d:owner&gt;&lt;cal:supported-calendar-component-set&gt;&lt;cal:comp name=&quot;VEVENT&quot;/&gt;&lt;/cal:supported-calendar-component-set&gt;&lt;cs:getctag&gt;169&lt;/cs:getctag&gt;&lt;d:resourcetype&gt;&lt;d:collection/&gt;&lt;cal:calendar/&gt;&lt;cs:shared-owner/&gt;&lt;/d:resourcetype&gt;&lt;d:supported-report-set&gt;&lt;d:supported-report&gt;&lt;d:report&gt;&lt;d:expand-property/&gt;&lt;/d:report&gt;&lt;/d:supported-report&gt;&lt;d:supported-report&gt;&lt;d:report&gt;&lt;d:principal-property-search/&gt;&lt;/d:report&gt;&lt;/d:supported-report&gt;&lt;d:supported-report&gt;&lt;d:report&gt;&lt;d:principal-search-property-set/&gt;&lt;/d:report&gt;&lt;/d:supported-report&gt;&lt;d:supported-report&gt;&lt;d:report&gt;&lt;cal:calendar-multiget/&gt;&lt;/d:report&gt;&lt;/d:supported-report&gt;&lt;d:supported-report&gt;&lt;d:report&gt;&lt;cal:calendar-query/&gt;&lt;/d:report&gt;&lt;/d:supported-report&gt;&lt;d:supported-report&gt;&lt;d:report&gt;&lt;cal:free-busy-query/&gt;&lt;/d:report&gt;&lt;/d:supported-report&gt;&lt;d:supported-report&gt;&lt;d:report&gt;&lt;d:principal-match/&gt;&lt;/d:report&gt;&lt;/d:supported-report&gt;&lt;d:supported-report&gt;&lt;d:report&gt;&lt;x:acl-query xmlns:x=&quot;urn:inverse:params:xml:ns:inverse-dav&quot;/&gt;&lt;/d:report&gt;&lt;/d:supported-report&gt;&lt;d:supported-report&gt;&lt;d:report&gt;&lt;x:user-query xmlns:x=&quot;urn:inverse:params:xml:ns:inverse-dav&quot;/&gt;&lt;/d:report&gt;&lt;/d:supported-report&gt;&lt;d:supported-report&gt;&lt;d:report&gt;&lt;cs:calendarserver-principal-search/&gt;&lt;/d:report&gt;&lt;/d:supported-report&gt;&lt;d:supported-report&gt;&lt;d:report&gt;&lt;d:sync-collection/&gt;&lt;/d:report&gt;&lt;/d:supported-report&gt;&lt;/d:supported-report-set&gt;&lt;/d:prop&gt;&lt;d:status&gt;HTTP/1.1 200 OK&lt;/d:status&gt;&lt;/d:propstat&gt;&lt;/d:response&gt;&lt;/d:multistatus&gt;

REQUEST: OPTIONS /calendars/2/ HTTP/1.1&quot; 200

# EMPTY RESPONSE #

REQUEST: PROPFIND /principals/users/2/ HTTP/1.1&quot; 207

&lt;?xml version=&quot;1.0&quot; encoding=&quot;utf-8&quot;?&gt;
&lt;d:multistatus xmlns:d=&quot;DAV:&quot; xmlns:s=&quot;http://sabredav.org/ns&quot; xmlns:card=&quot;urn:ietf:params:xml:ns:carddav&quot; xmlns:cal=&quot;urn:ietf:params:xml:ns:caldav&quot; xmlns:cs=&quot;http://calendarserver.org/ns/&quot; xmlns:owncloud=&quot;http://owncloud.org/ns&quot;&gt;&lt;d:response&gt;&lt;d:href&gt;/principals/users/2/&lt;/d:href&gt;&lt;d:propstat&gt;&lt;d:prop&gt;&lt;cal:calendar-home-set&gt;&lt;d:href&gt;/calendars/2/&lt;/d:href&gt;&lt;/cal:calendar-home-set&gt;&lt;cal:schedule-outbox-URL&gt;&lt;d:href&gt;/calendars/2/outbox&lt;/d:href&gt;&lt;/cal:schedule-outbox-URL&gt;&lt;cal:calendar-user-address-set&gt;&lt;d:href&gt;urn:uuid:2&lt;/d:href&gt;&lt;d:href&gt;mailto:jweberhofer@ADDRESS&lt;/d:href&gt;&lt;d:href&gt;/principals/users/2/&lt;/d:href&gt;&lt;/cal:calendar-user-address-set&gt;&lt;cal:schedule-inbox-URL&gt;&lt;d:href&gt;/calendars/2/inbox&lt;/d:href&gt;&lt;/cal:schedule-inbox-URL&gt;&lt;/d:prop&gt;&lt;d:status&gt;HTTP/1.1 200 OK&lt;/d:status&gt;&lt;/d:propstat&gt;&lt;/d:response&gt;&lt;/d:multistatus&gt;

REQUEST: REPORT /calendars/2/12/ HTTP/1.1&quot; 400
&lt;?xml version=&quot;1.0&quot; encoding=&quot;utf-8&quot;?&gt;
&lt;d:error xmlns:d=&quot;DAV:&quot; xmlns:s=&quot;http://sabredav.org/ns&quot;&gt;
  &lt;s:exception&gt;Sabre\DAV\Exception\BadRequest&lt;/s:exception&gt;
  &lt;s:message&gt;Invalid or unknown sync token&lt;/s:message&gt;
  &lt;s:sabredav-version&gt;1.8.10&lt;/s:sabredav-version&gt;
&lt;/d:error&gt;

**Comment by Gloirin** [comment]
**Comment posted by weberho on 20 Nov 2015 10:27**

It seems, as if support for sync-token has been added to sabredav with version 2.0 (http://sabre.io/dav/upgrade/1.8-to-2.0/ ) and tine has packaged version 1.8.10

**Comment by Gloirin** [comment]
**Comment posted by pmehrer on 20 Nov 2015 12:34**

thanks for the report.

Can you provide the body of the 4th request?

REQUEST: REPORT /calendars/2/12/ HTTP/1.1

you just provided the response body, but not the request body.

**Comment by Gloirin** [comment]
**Comment posted by weberho on 23 Nov 2015 11:20**

The request is:

&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;
&lt;sync-collection xmlns=&quot;DAV:&quot;&gt;&lt;sync-token/&gt;&lt;sync-level&gt;1&lt;/sync-level&gt;&lt;prop&gt;&lt;getcontenttype/&gt;&lt;getetag/&gt;&lt;/prop&gt;&lt;/sync-collection&gt;

**Comment by Gloirin** [comment]
**Comment posted by pschuele on 23 Nov 2015 14:51**

hi,

i just uploaded a patched file (Tinebase/WebDav/Plugin/SyncToken.php).

please test if this solves the problem.

**Comment by Gloirin** [comment]
**Comment posted by pschuele on 23 Nov 2015 14:52**

http://gerrit.tine20.com/customers/#/c/2415

**Comment by Gloirin** [comment]
**Comment posted by prego on 23 Nov 2015 14:54**

I can confirm that this patch works for me.

**Comment by Gloirin** [comment]
**Comment posted by weberho on 23 Nov 2015 15:06**

Philip, I can not access this patch; with http there is no access at all, without I see a message &quot;Configuration Error. Check the HTTP server&#039;s authentication settings.&quot; How can I access gerrit from outside your network?

**Comment by Gloirin** [comment]
**Comment posted by pschuele on 23 Nov 2015 15:20**

hi Johannes,

it&#039;s currently not possible to access our internal gerrit from outside, that&#039;s why I added this as a private comment just for reference.

I attached a fixed file to this issue:

2015-11-23 14:51	pschuele	File Added: SyncToken.php

**Comment by Gloirin** [comment]
**Comment posted by weberho on 23 Nov 2015 15:38**

Looks good! Sync is working again.

**Comment by Gloirin** [comment]
**Comment posted by weberho on 24 Nov 2015 08:21**

I found one calendar which still does not sync (it did with the older version). I could provide a tine20 loglevel 9 log via E-Mail. Thunderbird&#039;s log entry is:

Zeitstempel: 24.11.2015 09:02:07
Fehler: [calCachedCalendar] replay action failed: null, uri=https://SERVERNAME/tine20/calendars/c9710c87075a6f415c9564b8f43d489fceaf1578/9, result=2147500037, op=[xpconnect wrapped calIOperation]
Quelldatei: file:///home/web/.thunderbird/as77etml.default/extensions/%7Be2fda1a4-762b-4020-b5ad-a41df1933103%7D/calendar-js/calCachedCalendar.js
Zeile: 322

And Apache&#039;s log:

[24/Nov/2015:09:02:06 +0100] &quot;OPTIONS /tine20/calendars/c9710c87075a6f415c9564b8f43d489fceaf1578/ HTTP/1.1&quot; 200 - &quot;-&quot; &quot;Mozilla/5.0 (X11; Linux x86_64; rv:38.0) Gecko/20100101 Thunderbird/38.3.0 Lightning/4.0.3.1&quot;
[24/Nov/2015:09:02:06 +0100] &quot;PROPFIND /tine20/principals/users/c9710c87075a6f415c9564b8f43d489fceaf1578/ HTTP/1.1&quot; 207 1108 &quot;-&quot; &quot;Mozilla/5.0 (X11; Linux x86_64; rv:38.0) Gecko/20100101 Thunderbird/38.3.0 Lightning/4.0.3.1&quot;
[24/Nov/2015:09:02:06 +0100] &quot;REPORT /tine20/calendars/c9710c87075a6f415c9564b8f43d489fceaf1578/9/ HTTP/1.1&quot; 500 - &quot;-&quot; &quot;Mozilla/5.0 (X11; Linux x86_64; rv:38.0) Gecko/20100101 Thunderbird/38.3.0 Lightning/4.0.3.1&quot;

**Comment by Gloirin** [comment]
**Comment posted by weberho on 25 Nov 2015 14:59**

For the above mentioned calendar we have recognized, that we can add new calendar entries via Thunderbird even when the synchronization is broken and we do currently not see any of the entries in Thunderbird. The new entries are sent from thunderbird to tine.

**Comment by Gloirin** [comment]
**Comment posted by pschuele on 26 Nov 2015 08:02**

&gt; I found one calendar which still does not sync (it did with the older version). 
&gt; I could provide a tine20 loglevel 9 log via E-Mail. 

yes, please send the log (level 7 should be sufficient) to me (p.schueleATmetaways.de).

**Comment by Gloirin** [comment]
**Comment posted by pschuele on 27 Nov 2015 08:27**

received the log via mail.

thx @ weberho

**Comment by Gloirin** [comment]
**Comment posted by De Kus on 27 Nov 2015 09:47**

Installed update 2015.07.6-1.70 on Debian yesterday and my coworker experienced exactly this issue, while me did not. Today, after I had turned off my PC over night, I have the same issue. Yesterday we thought it might be the difference between him using Lightning 4.0 and me 4.4b (he uses Thunderbird, me SeaMonkey). So it was today I googled and found this ticket.

I immediately tried that fix and it worked for me without restarting anything. A logcheck using &#039;sudo tail -n 7000 /var/log/tine20/tine20-debug.log | grep &quot;Invalid or unknown sync token&quot;&#039; suggests that no new messages of that kind are occurring after replacing the file suggested in note 0017072 from pschuele.
